tests/test_data.py

```diff
@@ -500,6 +548,366 @@ def test_slicing(self):
             assert np.all(u.data[2:, 2:] == myrank)
             assert u.data[:2, 2:].size == u.data[2:, :2].size == u.data[:2, :2].size == 0
 
+        # Test slicing with a negative step
```

Comment:
this is huge. We need small unit tests, or better, as small as reasonably possible

this should then be split over multiple tests

Reply:
I just looked at the entire test :open_mouth:  "over many! multiple tests" :)